File: src/com/nayoshi12/christmasCracker/commands/MainExecutor.java
package com.nayoshi12.christmasCracker.commands;

import com.nayoshi12.christmasCracker.ChristmasCracker;
import com.nayoshi12.christmasCracker.commands.subcommands.GiveItemCommand;
import com.nayoshi12.christmasCracker.commands.subcommands.PluginReloadCommand;
import net.md_5.bungee.api.chat.ClickEvent;
import org.bukkit.ChatColor;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import java.awt.*;
import java.util.ArrayList;
import java.util.List;

/**
 * Created by Matthew on 12/19/2016.
 */
public class MainExecutor implements CommandExecutor{
    List<SubCommand> sub = new ArrayList<>();
    ChristmasCracker pl;
    public MainExecutor(ChristmasCracker pl){
        this.pl = pl;
        sub.add(new PluginReloadCommand(pl,"reload"));
        sub.add(new GiveItemCommand(pl,"give"));
    }
    @Override
    public boolean onCommand(CommandSender sender, Command cmd, String s, String[] args) {
        if(cmd.getName().equals("christmascracker")) {

            if(args.length != 0) {
                for(SubCommand cmdSub:sub){
                    if(args[0].equalsIgnoreCase(cmdSub.getName())){
                        cmdSub.execute(sender,args);
                    }
                }
            } else {
                if(sender instanceof Player) {
                    sender.sendMessage(""+ ChatColor.GREEN +""+ ChatColor.STRIKETHROUGH +"-------------------------");
                    String text = (ChatColor.GREEN + "Made with love by ");

                    String name = ChatColor.GREEN + ""+ ChatColor.BOLD+"Redrield and Nayoshi";
                    ClickEvent same = new ClickEvent(ClickEvent.Action.OPEN_URL, "https://redrield.com");
                    text += name;
                    sender.sendMessage(text);
                    sender.sendMessage(ChatColor.GREEN + "" + ChatColor.STRIKETHROUGH + "-------------------------");
                }
            }
            return true;
        }
        return false;
    }
}

File: src/main/java/com/redrield/christmascracker/commands/MainExecutor.kt
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

package com.redrield.christmascracker.commands

import com.redrield.christmascracker.ChristmasCracker
import com.redrield.christmascracker.commands.subcommands.GiveItemCommand
import com.redrield.christmascracker.commands.subcommands.PluginReloadCommand
import com.redrield.christmascracker.plus
import net.md_5.bungee.api.chat.ClickEvent
import net.md_5.bungee.api.chat.TextComponent
import org.bukkit.ChatColor
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player
import java.util.ArrayList

class MainExecutor(val plugin: ChristmasCracker) : CommandExecutor {

    val subCommands = ArrayList<SubCommand>()

    init {
        subCommands.add(PluginReloadCommand(plugin))
        subCommands.add(GiveItemCommand(plugin))
    }

    override fun onCommand(cs: CommandSender, cmd: Command, label: String, args: Array<String>): Boolean {
        if(cmd.name.equals("christmascracker", true)) {

                if(args.isNotEmpty()) {
                    when(args[0].decapitalize()) {
                        "reload" -> {
                            subCommands.filter { it.name == "reload" }.single().execute(cs, args)
                        }
                        "give" -> {
                            subCommands.filter { it.name == "give" }.single().execute(cs, args.sliceArray(1..args.size-1))
                        }
                    }
                } else {
                    if(cs is Player) {
                        cs.sendMessage("${ChatColor.GREEN}${ChatColor.STRIKETHROUGH}-------------------------")
                        val text = TextComponent(ChatColor.GREEN + "Made with love by ")
                        val name = TextComponent("${ChatColor.GREEN}${ChatColor.BOLD}Redrield")
                        name.clickEvent = ClickEvent(ClickEvent.Action.OPEN_URL, "https://redrield.com")
                        text.addExtra(name)
                        cs.spigot().sendMessage(text)
                        cs.sendMessage("${ChatColor.GREEN}${ChatColor.STRIKETHROUGH}-------------------------")
                    }
                }
            return true
        }
        return false
    }
}
